lib/jekyll/hooks.rb

```diff
@@ -88,19 +80,17 @@ def self.insert_hook(owner, event, priority, &block)
     end
 
     # interface for Jekyll core components to trigger hooks
-    def self.trigger(instance, event, *args)
-      owner_symbol = OWNER_MAP[instance.class]
-
+    def self.trigger(owner, event, *args)
```

Comment:
The cleanest and most direct IMO is option one.  It's not just that area we had trouble, it was within Jekyll itself too.  The problem within Jekyll's own source is what actually started this entire conversation weeks before this one spawned the change.

Where at one point we were trying to add in hooks for Jekyll's internal configurations so we could hook out dozens of points, we couldn't get a subclass because it was Hash because the code did not speak for itself. That's too broad, so we needed to make things open and explicit in that a user should have to pass an owner and a hook, this makes it so that anybody can call any hook at any point for any reason.

It's also not just us we need to worry about, it's plugins too, the hook system is only as valuable as it's usage so we have really two problems here:

1. How do we trigger hooks.
2. How do we let plugins have their own hooks so this: https://github.com/jekyll-assets/jekyll-assets/blob/master/lib/jekyll/assets/hook.rb doesn't have to happen.

IMO there should be no basis for resistance against changing the API to the more explicit version because hooks haven't been around long enough to gain usage, there was never any option for plugins to use them directly (which means a plugin can't be broken) and the messaging to hooks was only used internally so this shouldn't in theory and in testing break anything like Octopress, because it's only asking to add a hook to a point... we are the ones who are messaging the hooks on the point... not Octopress.